src/test/TestAccount.h

```diff
@@ -132,6 +132,10 @@ class TestAccount
     nextSequenceNumber()
     {
         updateSequenceNumber();
+        if (mSn == std::numeric_limits<SequenceNumber>::max())
```

Comment:
why not throw instead? This seems to hide a bug in whatever test does this

Reply:
It's the "large bump" BumpSequence test, as mentioned in the description.  This is what it's expecting:

```
            SECTION("large bump")
            {
                auto newSeq = INT64_MAX;
                a.bumpSequence(newSeq);
                REQUIRE(a.loadSequenceNumber() == newSeq);
                SECTION("no more tx when INT64_MAX is reached")
                {
                    REQUIRE_THROWS_AS(a.pay(root, 1), ex_txBAD_SEQ);
                }
            }
```

So I can make that pass by throwing `txtest::ex_txBAD_SEQ{}` in the code above, although it would mean that the test wasn't testing what it was trying to anymore -- I think it wants to go through `TestAccount::pay()` -> `applyTx()` -> `throwIf()` to confirm that the transaction containing the payment operation fails with `txBAD_SEQ`.  Specifically I think it's trying ultimately to exercise this:

```
bool
TransactionFrame::isBadSeq(int64_t seqNum) const
{
    return seqNum == INT64_MAX || seqNum + 1 != getSeqNum();
}
```

If we `throw` in `nextSequenceNumber()` then I think this test will just be testing test code.

`TestAccount::tx()` does have an optional `SequenceNumber` parameter; maybe we should just use it to create a `TransactionFrame` directly in the `"large bump"` and call `applyTx()` on it ourselves?

Reply:
> TestAccount::tx() does have an optional SequenceNumber parameter; maybe we should just use it to create a TransactionFrame directly in the "large bump" and call applyTx() on it ourselves?

This looks reasonable to me and fixes the test, so I've done it.

Reply:
yes fixing the test is the way to go